sympy/matrices/matrices.py

```diff
@@ -2357,22 +2358,28 @@ def __ne__(self, other):
         return not self == other
 
     def _matrix_pow_by_jordan_blocks(self, num):
-        from sympy.matrices import diag, MutableMatrix
+        from sympy.matrices import diag, MutableMatrix, eye
         from sympy import binomial
 
         def jordan_cell_power(jc, n):
             N = jc.shape[0]
             l = jc[0, 0]
-            if l == 0 and (n < N - 1) != False:
-                raise ValueError("Matrix det == 0; not invertible")
-            elif l == 0 and N > 1 and n % 1 != 0:
-                raise ValueError("Non-integer power cannot be evaluated")
-            for i in range(N):
-                for j in range(N-i):
-                    bn = binomial(n, i)
-                    if isinstance(bn, binomial):
-                        bn = bn._eval_expand_func()
-                    jc[j, i+j] = l**(n-i)*bn
+            if l.is_zero:
+                if N == 1 and n.is_nonnegative:
+                    jc [0, 0] = l**n
```

Comment:
I find this spacing like `jc [0, 0]` or `range (N)` odd. Normal Python style doesn't add spaces there. I don't think it's consistent with the rest of the file either.

Reply:
My style creeping in, I try not to...